daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -231,7 +225,6 @@ abstract class OrderedSequenceParserBase(
                           // discarded.
                           //
                           if (!isFirstIteration) pstate.discard(priorState)
```

Comment:
Should this just always discard or reset to priorState?

Reply:
oops. yes I think so.